core/string/ustring.cpp

```diff
@@ -162,6 +162,18 @@ bool CharString::operator<(const CharString &p_right) const {
 	return is_str_less(get_data(), p_right.get_data());
 }
 
+bool CharString::operator==(const char *p_right) const {
+	if (length() == 0) {
+		// True if both have length 0, false if only p_right has a length.
+		return !p_right;
```

Comment:
Is `""` actually an empty pointer in c++? Otherwise this will fail in the case when checked against that

Reply:
Ah yeah good point, I might have to check for the `CharString` storing NULL input as a single null terminator (not actually sure what it does off hand, but I'll double check this.

Reply:
Would check if `p_right` is `nullptr` or if `!p_right[0]`

Reply:
Length excludes the null terminator: `int length() const { return size() ? size() - 1 : 0; }`

Reply:
Maybe:
```cpp
if (p_right == nullptr || p_right[0] == 0) {
    return length() == 0;
} else if (length() == 0) {
    return false;
}
return strcmp(ptr(), p_right) == 0;
```